library/src/main/java/com/google/android/exoplayer/hls/HlsChunkSource.java

```diff
@@ -350,6 +384,7 @@ public void selectTrack(int index) {
     selectedVariantIndex = selectedTrack.defaultVariantIndex;
     variants = selectedTrack.variants;
     variantPlaylists = new HlsMediaPlaylist[variants.length];
+    availableRange = new TimeRange[variants.length];
```

Comment:
currentAvailableRangeIndex must be reset to -1 here otherwise when the user selects a specific track after starting the playback, it will throw out of array exception.